### PR TITLE
Add hive-style path attributes to directory spools

### PR DESCRIPTION
**Pull request by d-chambers**
## Description

Directory spools now parse `key=value` pairs out of the paths inside the spool, Hive-style. Each directory segment can hold one pair (`network=XX/`), and any segment — including the file name, with its extension stripped — can hold several separated by `__` (the same separator `get_patch_names` uses), e.g. `station=A__tag=raw.h5`. Parsed values become string attrs: they appear in `get_contents()`, work with `select` (equality, collections, glob, regex), participate in chunk grouping, survive unions/pickling, and are stamped onto loaded patches. When a path attr and a file-declared attr share a name, the path wins — renaming a directory (or file) attaches or corrects metadata without rewriting data.

Details:

- Parser: `dascore.utils.paths.parse_hive_path_attrs` (percent-decoding, deepest-key-wins, Hive NULL sentinel skipped, letter-led extension stripping so `depth=1.5` survives). Reserved/underscore key names follow the same skip/warn rules as file attrs.
- Provenance: the applied dict is persisted per source (`sources.path_attrs`, JSON) and surfaced privately as `_path_attrs`, since derived/union catalogs absolutize paths. `INDEX_VERSION` 3 → 4; existing indexes rebuild automatically.
- Move detection: updates match a disappeared path to a brand-new one by exact `(mtime_ns, size_bytes)` identity (unique 1:1 on both sides; the rename-invariant manifest signature for directory-format units) and rewrite paths + path-derived attrs in batched SQL, never re-reading contents. Renames that *remove* a hive key, or ambiguous stat matches, fall back to a rescan.
- Values stay strings (no type inference); numeric inference could be a follow-up if range selects on path keys turn out to matter.

Benchmarks vs `dev` (600 files / 12 partition dirs): cold index build, no-op update, time select, and patch loading unchanged within noise; renaming one partition dir + update went 457 ms → 72 ms, renaming all 12 dirs 4.13 s → 0.12 s (~34×, and the gap grows with archive size since dev rescans contents). Path parsing costs ~1.5 µs/path. Two codspeed benchmarks added (`TestHivePathAttrBenchmarks`: hive indexing, patch loading with stamped attrs).

Behavior note for release: directories whose names happen to contain `=` now silently produce attrs, and the version bump forces a one-time index rebuild.

## Changelog

- added: directory spools parse Hive-style `key=value` path segments into string attrs, usable in `get_contents()`, `select`, and chunk grouping; a path attr wins over a file-declared attr of the same name, so renaming a directory attaches metadata without rewriting data. Any path segment containing `=` is parsed this way, so a directory named for something else that happens to contain one now yields an attr.

## Checklist

I have (if applicable):

- [ ] referenced the GitHub issue this PR closes.
- [x] documented the new feature with docstrings and/or appropriate doc page.
- [x] included tests. See [testing guidelines](https://dascore.org/contributing/testing.html).
- [x] added the "ready_for_review" tag once the PR is ready to be reviewed.

## Summary by CodeRabbit

* **New Features**
  * Added support for Hive-style `key=value` path attributes in directory-based spools.
  * Path attributes are now stored and surfaced in spool contents, enabling filtering and appearing on loaded patches.
  * Includes correct parsing (encoded values, `__`-separated pairs, repeated-key precedence) and efficient rename/move handling that updates attributes without rescanning when safe.
* **Bug Fixes**
  * Path-derived attributes correctly override conflicting file-defined attributes.
* **Documentation**
  * Expanded spool index and tutorial docs with examples and rename edge cases.
* **Tests**
  * Added end-to-end coverage for parsing, selection, patch stamping, and move detection.

### CODE REVIEW

File: dascore/io/index/ingest.py [hunk 1]
```diff
@@ -198,31 +201,76 @@ def typed_value(value) -> TypedValue | None:
...
+    """
+    out = {}
+    for name, value in parse_hive_path_attrs(rel_posix).items():
+        status = _attr_name_status(name)
+        if status == "reserved" and warn:
+            msg = (
+                f"Skipping hive-style path key {name!r} in {rel_posix!r}; "
+                "it collides with a structural index column."
+            )
+            warnings.warn(msg, UserWarning)
```
[the remark this answers is not in the document]
Comment (d-chambers) [reply dascore/io/index/ingest.py]: Done in 95d21f2b — both warnings now pass stacklevel=2, matching backend.py.
